plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -452,302 +240,403 @@ private void createOpenShiftApplicationControls(FormToolkit toolkit, Composite p
 	private void createPodPathControls(Composite parent, DataBindingContext dbc) {
 		Label deployPathLabel = new Label(parent, SWT.NONE);
 		deployPathLabel.setText("Pod Deployment Path: ");
-		GridDataFactory.fillDefaults()
-			.align(SWT.BEGINNING, SWT.CENTER)
-			.applyTo(deployPathLabel);
-
-		Text deployPathText = new Text(parent, SWT.BORDER);
-		GridDataFactory.fillDefaults()
-			.align(SWT.FILL, SWT.CENTER)
-			.span(2, 1).grab(true, false)
-			.applyTo(deployPathText);
-		ValueBindingBuilder
-			.bind(WidgetProperties.text(SWT.Modify).observe(deployPathText))
-			.validatingAfterConvert(new IValidator() {
-
-				@Override
-				public IStatus validate(Object value) {
-					if (!(value instanceof String) || StringUtils.isEmpty(value)) {
-						return ValidationStatus.cancel("Please provide a path to deploy to on the pod.");
-					}
-					if (!Path.isValidPosixPath((String)value)) {
-						return ValidationStatus.error("Please provide a valid path to deploy to on the pod");
-					}
-					return ValidationStatus.ok();
+		applyAlignData(SWT.BEGINNING, SWT.CENTER,deployPathLabel);
+
+		deployPathText = new Text(parent, SWT.BORDER);
+		applyGridData(2, 1, SWT.FILL, SWT.CENTER,true, false,deployPathText);
+		
+		deployPathModifyListener = new ModifyListener() {
+			public void modifyText(ModifyEvent e) {
+				execute(new ServerWorkingCopyPropertyTextCommand(input.getServer(), "Set Deploy Path", deployPathText, 
+						deployPathText.getText(), OpenShiftServerUtils.ATTR_POD_PATH, this));
+			}
+		}; 
+		ModifyListener deployPathDecoration = new TextboxDecoratorSupport(deployPathText, new IValidator() {
+			@Override
+			public IStatus validate(Object value) {
+				// TODO validate 
+				if (!(value instanceof String) || StringUtils.isEmpty(value)) {
+					return ValidationStatus.cancel("Please provide a path to deploy to on the pod.");
+				}
+				if (!Path.isValidPosixPath((String)value)) {
+					return ValidationStatus.error("Please provide a valid path to deploy to on the pod");
```

Comment:
can be written in a more jdk8 like fashion with lambda:
```
		ModifyListener deployPathDecoration = new TextboxDecoratorSupport(deployPathText, 
			value -> {
				// TODO validate 
				if (!(value instanceof String) || StringUtils.isEmpty(value)) {
					return ValidationStatus.cancel("Please provide a path to deploy to on the pod.");
				}
				if (!Path.isValidPosixPath((String)value)) {
					return ValidationStatus.error("Please provide a valid path to deploy to on the pod");
				}
				return Status.OK_STATUS;
		});
```
like many of those listeners that you use.